File: web/src/wire/conversation.ts
import type { Conversation } from "../context/ChatContext";

export type ConvDto = {
  id: string;
  type: "group" | "direct";
  title:  string;
  avatar_url?: string | null;
  last_msg_id?: string;
  last_msg_text?: string;
  last_msg_from?: string;
  last_msg_ts?: number;
  last_seq: number;
  last_read_seq: number;
  unread_count: number;
};

export const toConversation = (w: ConvDto): Conversation => ({
  id: w.id,
  type: w.type,
  title: w.title,
  avatarUrl: w.avatar_url ?? null,
  lastMsgId: w.last_msg_id,
  lastMsgText: w.last_msg_text,
  lastMsgFrom: w.last_msg_from,
  lastMsgTs: w.last_msg_ts,
  lastSeq: w.last_seq,
  lastReadSeq: w.last_read_seq,
  unreadCount: w.unread_count,
});

File: backend/wire/conversation.go
package wire

import (
	"github.com/google/uuid"
)

type ParticipantSummary struct {
	UID       uuid.UUID `json:"uid"`
	Name      string    `json:"name"`
	Role      string    `json:"role"` // "admin" | "member"
	AvatarURL string    `json:"avatar_url"`
}

// conversation summary
type ConversationSummary struct {
	ConversationID uuid.UUID `json:"id"`
	ConvType       string    `json:"type"` // group | direct

	Title       string    `json:"title,omitempty"`
	AvatarURL   string    `json:"avatar_url,omitempty"`
	LastMsgID   uuid.UUID `json:"last_msg_id,omitempty"`
	LastMsgText string    `json:"last_msg_text,omitempty"`
	LastMsgFrom uuid.UUID `json:"last_msg_from,omitempty"`
	LastMsgTs   int64     `json:"last_msg_ts,omitempty"`

	LastSeq     int64 `json:"last_seq"`
	LastReadSeq int64 `json:"last_read_seq"`
	UnreadCount int64 `json:"unread_count"`

	Participants []ParticipantSummary `json:"participants"`
}
